extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -848,6 +902,143 @@ private PushdownResult pushdownRowGroupMetadata(
         }
     }
 
+    /**
+     * Apply the chunk filter to the row group dictionaries and return the result.
+     */
+    @NotNull
+    private PushdownResult pushdownFilterDictionary(
+            final RowSet selection,
+            final RegionedColumnSourceManager.RegionedColumnSourcePushdownFilterContext ctx,
+            final String[] parquetColumnNames,
+            final PushdownResult result) {
+
+        final BasePushdownFilterContext.FilterNullBehavior filterNullBehavior = ctx.filterNullBehavior();
+        if (filterNullBehavior == BasePushdownFilterContext.FilterNullBehavior.FAILS_ON_NULLS) {
+            // We cannot use dictionary-based filtering, keep all the rows as "maybe" rows. Later when the
+            // filter is applied to actual data, it will NPE if there are null values in the table.
+            return result.copy();
+        }
```

Comment:
I need to think a bit deeper on this. It still confuses me that the methods have access to both `selection` and `result`.